provisioner/src/lib.rs

```diff
@@ -236,6 +258,294 @@ impl MyProvisioner {
         }
     }
 
+    async fn get_prefix(&self, project_name: &str) -> String {
+        let mut hasher = Sha256::new();
+
+        hasher.update(project_name.as_bytes());
+
+        let hash = hasher.finalize();
+
+        // 43 characters long (4 characters correspond to 3 bytes of data)
+        // sha256 is 32 bytes. 32 / 3 * 4 ~ 43
+        // we care about this because various AWS identifiers have specific length constraints
+        Base64UrlUnpadded::encode_string(&hash)
+    }
+
+    async fn get_dynamodb_policy_name(&self, prefix: &str) -> String {
+        format!("{}-dynamo-policy", prefix)
+    }
```

Comment:
These methods don't use `&self`, maybe they should be outside of the struct `impl`?